### ISSUE TITLE
Is the clickhouse-rs client thread safe?

### ISSUE THREAD

**Issue by monadbobo** [suharev7/clickhouse-rs#179]
For example if I use the same pool object in multiple tokio tasks will there be undefined behavior.

Or should I share the connection through Arc<pool> in different tasks, or should each task call pool.clone() to share the connection。

**Comment by suharev7** [comment]
Yes, it's thread safe. 'pool.clone()' and sharing the connection through Arc will both work correctly.
